src/chaat/handler/handler.clj

```diff
@@ -0,0 +1,52 @@
+(ns chaat.handler.handler
+  (:require [ring.util.response :as res]
+            [chaat.model.user :as model.user]
+            [java-time.api :as jt]
+            [chaat.handler.validation :as handler.validation]
+            [chaat.errors :refer [do-or-error]]
+            [chaat.db.utils :as db.utils]))
+
+(defn home
+  "Respond with a greeting/welcome"
+  [request]
+  (res/response "Welcome to chaat"))
+
+(defn health-check
+  "Respond with application status: checks server and db status"
+  [db request]
+  (let [result (db.utils/health-check db)]
+    (if (nil? (:error result))
+      (res/response (str "all systems go: " (jt/instant)))
+      (res/bad-request (str (:error result))))))
+
+(defn signup
+  "Create a user account with supplied parameters"
+  [db request]
+  (let [params (:params request)
+        {:keys [username password]} params
+        result (handler.validation/validate-signup-details username password)
+        result (do-or-error result model.user/create db username password)]
+    (if (nil? (:error result))
+      (res/response "Signup successful")
+      (res/bad-request (str (:error result))))))
+
+(defn delete-user
+  "Delete a user account"
+  [db request]
+  (let [params (:params request)
+        username (:username params)
+        result (handler.validation/validate-username username)
+        result (do-or-error result model.user/delete db username)]
+    (if (nil? (:error result))
+      (res/response (str request))
+      (res/bad-request (str (:error result))))))
```

Comment:
[style nit] In general, I prefer error cases first, so I'd have inverted this `if`. Regardless, since you need to check various types of errors, consider `case`, something like:
```suggestion
    (case (get-error result)
      nil              (res/response (str request))
      :invalid-request (res/bad-request (str (:error result)))
      (status 500))))
```

The above interface would be easier to achieve if you have a standard machine-readable representation for errors along with the strings.

Reply:
Thanks for the suggested change, `case` would be very useful here.
I had already started making a helper function `send-response` to do this, but will keep this approach in mind.